### DISCUSSION TITLE
When importing Roam EDN: add the option to download all images from Firebase and link to them locally

### DISCUSSION THREAD

**Discussion by NishantTharani** [athensresearch/athens#1332] [Feature Requests]
It's great to see the Roam import functionality implemented recently! One of the main draws of Athens for me is the privacy-friendly ability to have everything hosted locally. It would be great if the Roam importer could be extended to make this true for images as well.

Roam's approach to images is that whenever you paste an image into a note, the image is uploaded to Firebase, and a link to the image is generated and inserted into your note. The link is publicly accessible. This means that when a Roam EDN is imported into Athens currently, the Firebase link is preserved, so the image displayed in Athens is still pulled from that link.

I would like to suggest that the import process includes a step that downloads all the images from these Firebase links and saves them locally in the ./images subfolder, then links to them there.

There is a topic on the Roam forums where someone has posted some quick code that does this for the markdown export: https://forum.roamresearch.com/t/download-files-uploaded-to-firebase/3808 - the first script looks for matching URLs and downloads each file once, the second script replaces the markdown links with the local file locations. 

I assume this downloading would be OK with Roam, because the images were initially uploaded by the user doing the export/import, so they have the right to access them? It also seems like this would save them bandwidth in the long run, because the Athens user wouldn't be using their firebase each time they view the image.

[a comment by an automated account is not shown]

**Comment by sid597** [comment]
@NishantTharani The forum link you provided does not work.

**Reply by NishantTharani** [reply]
@sid597   Thank you, this is probably because I posted the issue a while ago and the forum site may have gone down in the meantime.

However, could I ask why my issue was closed? The forum link is only an option for starting with a solution, but it is merely incidental to the feature request - it seems like any import from roam research that is designed to include pictures and not just text should perform this. Could the issue be reopened please? Edit: or is this the correct place for feature requests ('discussions', not 'issues')? Thanks!

**Reply by sid597** [reply]
@NishantTharani Thanks for clarifying that, as you said this is a feature request so moved this to discussions. We are currently in the process of triaging issues, this will be moved to issues when either this post gets enough votes or is in our roadmap. Please checkout #1055

**Reply by NishantTharani** [reply]
@sid597 Got it, thank you! Another related issue I opened at the same time is #927 - this one seems to me like an actual bug and not a feature request. I guess it will get triaged in due course. 

I think it's possible the forum link will work again as maybe the site is just temporarily down. If so I'll post here again.

I would like to suggest that if this is seen as a feature request and not a bug, then the roam import feature should be clarified to say that it is only importing text content, not media content. With the feature as it is, the media content would still be hosted by Roam Research and under their control, not stored locally/under the user's control. 

Furthermore, Roam now offers the option to delete media files you uploaded: a common user workflow would probably be to import their Roam graph into Athens and then delete the Roam graph, thinking it's safe to do now that it's been imported, but then they might lose all their images. It might be worth clarifying that point if this is not going to be a bug?

**Reply by sid597** [reply]
@NishantTharani I think #927 is a bug and it should remain in issues. Could you please copy-paste this comment there also, in #927 I can link this discussion but I think it would be very helpful to get all info in one place.

**Reply by NishantTharani** [reply]
@sid597 Thanks! My comment was actually related to #928 though. To clarify:

* #927 : Not related to roam research import at all, a bug related to moving the Athens file system not updating image links, we agree this is a bug
* #928 : Related to roam research import, images aren't downloaded and saved locally when import is run, so if user deletes their roam media/graph after importing they will lose their images. I think this is a bug, if not fixed then a warning should be added to the 'Import from Roam' feature to make it clear that only text will be imported, not images. What do you think?

**Reply by sid597** [reply]
@NishantTharani I saw "images" in #927 and in your comments and thought you were talking about Roam research images 😅. I think #927 is solved can you please check, let's carry on this in #927.

> #928 : Related to roam research import, images aren't downloaded and saved ...

 I agree with what you said here, it is a bug I think you should update #928 with what you described here and the current 928 can be a suggested solution (which makes sense). Reopening 928 in issues.

**Reply by sid597** [reply]
@NishantTharani It is not possible to revert this to issues, a new has to be created https://github.com/github/feedback/discussions/2835 should I create a new issue or can you do that?

**Reply by NishantTharani** [reply]
@sid597 I will create a new issue and tag you in it

**Reply by sid597** [reply]
I misunderstood #927 is not solved.

**Reply by NishantTharani** [reply]
Recreated #928 in #1334
